.github/workflows/ci.yml

```diff
@@ -32,19 +32,26 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
+        uses: actions/setup-python@9e62be81b28222addecf85e47571213eb7680449
         with:
           python-version: ${{ matrix.python-version }}
           architecture: ${{ matrix.architecture }}
 
+
+      - if: ${{ endsWith(matrix.python-version, 't') }}
+        run:
+          pip install --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
```

Comment:
This can be left out, since `numpy` 2.2.4 has stable support for free-threading. You can just drop this line I believe.